ledger/participant-integration-api/src/main/scala/platform/store/backend/common/ContractStorageBackendTemplate.scala

```diff
@@ -64,26 +66,23 @@ trait ContractStorageBackendTemplate extends ContractStorageBackend {
         import com.daml.platform.store.Conversions.ContractIdToStatement
         SQL"""
   WITH archival_event AS (
-         SELECT participant_events.*
-           FROM participant_events, parameters
+         SELECT 1
+           FROM participant_events_consuming_exercise, parameters
```

Comment:
Wasn't the original motivation to limit changes to H2? This changes for all...

Reply:
I may have prematurely moved this to the common class following discussions in this PR. After a second thought I would go for having changes only for H2, especially that it's close to a new release.